demo/src/main/java/com/smashingboxes/surelockdemo/LoginActivity.kt

```diff
@@ -0,0 +1,346 @@
+                preferences = getSharedPreferences(SHARED_PREFS_FILE_NAME, Context.MODE_PRIVATE)
+            }
+            return preferences!!
+        }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+        surelockStorage = SharedPreferencesStorage(this, SHARED_PREFS_FILE_NAME)
+        setUpViews()
+    }
+
+    private fun setUpViews() {
+        password.setOnEditorActionListener(
+            TextView.OnEditorActionListener { _, id, _ ->
+                if (id == R.id.login || id == EditorInfo.IME_NULL) {
+                    handleLoginClick()
+                    return@OnEditorActionListener true
+                }
+                false
+            })
+
+        val signInButton = findViewById<View>(R.id.sign_in_button) as Button
+        signInButton.setOnClickListener { handleLoginClick() }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (userWantsToUseFingerprintToLogin()) {
+            if (!Surelock.fingerprintAuthIsSetUp(this, true)) {
+                fingerprint_checkbox.visibility = View.VISIBLE
+                fingerprint_checkbox.isChecked = false
+            } else {
+                try {
+                    getSurelock().loginWithFingerprint(KEY_CRE_DEN_TIALS)
+                } catch (e: SurelockInvalidKeyException) {
+                    surelockStorage.clearAll()
+                    showFingerprintLoginInvalidated()
+                }
+
+                fingerprint_checkbox.visibility = View.GONE
+            }
+        }
+    }
+
+    override fun onStop() {
+        //Clear storage so users don't have to clear app data to try out new credentials
+        surelockStorage.clearAll()
+        super.onStop()
+    }
+
+    private fun getSurelock(): Surelock {
+        if (surelock == null) {
+            surelock = Surelock.Builder(this)
+                .withDefaultDialog(R.style.SurelockDemoDialog)
+                .withKeystoreAlias(KEYSTORE_KEY_ALIAS)
+                .withFragmentManager(fragmentManager)
+                .withSurelockFragmentTag(FINGERPRINT_DIALOG_FRAGMENT_TAG)
+                .withSurelockStorage(surelockStorage)
+                .build()
+        }
+        return surelock!!
+    }
+
+    private fun showFingerprintLoginInvalidated() {
+        val builder = AlertDialog.Builder(this)
+        builder.setTitle(R.string.sl_login_error)
+            .setMessage(R.string.sl_re_enroll)
+            .setPositiveButton(android.R.string.ok) { dialog, _ -> dialog.dismiss() }.show()
+    }
+
+    private fun userWantsToUseFingerprintToLogin(): Boolean {
+        return preferenceHelper.contains(KEY_CRE_DEN_TIALS)
+    }
+
+    private fun validateLoginForm(usernameInput: String?, passwordInput: String?): Boolean {
+        var errorsExist = false
+        var focusView: View? = null
+
+        if (TextUtils.isEmpty(usernameInput)) {
```

Comment:
Kotlin has a `isNullOrEmpty()` extension for Strings.
You should use that instead of TextUtils nowadays.